src/main/webapp/app/pages/usageAnalysisPage/UsageAnalysisPage.tsx

```diff
@@ -0,0 +1,172 @@
+import { action, observable } from 'mobx';
+import { inject, observer } from 'mobx-react';
+import { RouterStore } from 'mobx-react-router';
+import React from 'react';
+import Client from 'app/shared/api/clientInstance';
+import { match } from 'react-router';
+import { notifyError } from 'app/shared/utils/NotificationUtils';
+import Tabs from 'react-bootstrap/Tabs';
+import Tab from 'react-bootstrap/Tab';
+import OncoKBTable from 'app/components/oncokbTable/OncoKBTable';
+import { filterByKeyword } from 'app/shared/utils/Utils';
+import { UsageRecord, UserUsageOverview } from 'app/shared/api/generated/API';
+import _ from 'lodash';
+import { Link } from 'react-router-dom';
+import autobind from 'autobind-decorator';
+import { ToggleButton, ToggleButtonGroup } from 'react-bootstrap';
+import UsageDetailTable from './UsageDetailTable';
+import { USAGE_TOP_USERS_LIMIT } from 'app/config/constants';
+
+@inject('routing')
+@observer
+export default class UsageAnalysisPage extends React.Component<{
+  routing: RouterStore;
+  match: match;
+}> {
+  @observable users: UserUsageOverview[] = [];
+  @observable loadedUsers = false;
+  @observable toggleValue = 1;
+
+  @observable loadedResources = false;
+  @observable usageDetail: Map<string, UsageRecord[]> = new Map();
+  @observable dropdownList: string[] = [];
+
+  constructor(props: Readonly<{ routing: RouterStore; match: match }>) {
+    super(props);
+    this.getUsers();
+    this.getResource();
+  }
+
+  @action
+  async getUsers() {
```

Comment:
This is ok, but if the page does not have interactive feature that requires loading data constantly for the same data field, I would just use `remoteData` as much as possible so you have a better status attached. Then you would not need the `loadedUsers` anymore.